apps/docs-snippets/src/guide/predicates/predicate-with-configurable.test.ts

```diff
@@ -28,10 +28,13 @@ describe(__filename, () => {
 
     const configurable = { WHITELISTED: newWhitelistedAddress };
     // instantiate predicate with configurable constants
-    const predicate = new Predicate(bin, wallet.provider, abi, configurable);
-
-    // set predicate data to be the same as the configurable constant
-    predicate.setData(configurable.WHITELISTED);
+    const predicate = new Predicate({
+      bytecode: bin,
+      provider: wallet.provider,
+      abi,
+      inputData: [configurable.WHITELISTED],
+      configurableConstants: configurable,
+    });
```

Comment:
Maybe we could use the generic in `Predicate<T>` for demo purposes if that makes sense.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1826/commits/db8be677225d5f6be0a984f75d3f71fa78405a2e